### ISSUE TITLE
Variation track data

### ISSUE THREAD

**Issue by Antonialock** [pombase/website#729]
Outstanding: Colour variants by type if possible

**Comment by ValWood** [comment]
In the ensembl browser.....different types of variation have different colour...

![variant](https://user-images.githubusercontent.com/7359272/38836719-20858b0e-41c7-11e8-8607-e52f61fbe58b.jpg)

details of the SNP appear in the pop-up.

Not sure if these are available in JBrowse.
For  a while we might need to replace the Ensembl link... if we can't host this dataset easily in the next month or so we could tell people that it is still available.

**Comment by ValWood** [comment]
it might not be a bad idea to have a permanent "external link" to variation data for a gene too.

the Ensembl view for this is quite good,

http://fungi.ensembl.org/Schizosaccharomyces_pombe/Location/View?db=core;g=SPCC18B5.03;r=III:721788-725835;t=SPCC18B5.03.1

each variant links to this info (which I resume is pulled from the variation database)

![variation](https://user-images.githubusercontent.com/7359272/38850650-46af502e-420a-11e8-8e77-a9228d4635eb.jpg)

**Comment by ValWood** [comment]
(although a link from JBrowse to the appropriate SNP in the DB will do  for this meta data I have no idea how this connection is made)

**Comment by kimrutherford** [comment]
I've had a look at the variation data files found by Midori.  We'll need to fix the chromosome IDs in the VCF files before we can load them into JBrowse.  It's currently I, II, III etc.

Also there seem to be some tabs missing for the mitochrodrial data in both files.  It looks like there should be tabs after the "MTR" in the first column:

    III     2431124 .       C       CTGCATAGCATCAAACCGAGAAT 106253  MEAN_ALLE
    III     2431989 .       ACT     A       6982.84 DP;MEAN_ALLELE_BALANCE  A
    III     2433201 .       T       TG      118279  MQ      AC=49;ACMI=8;ACMJ
    MTR5053 UNION_BC_k31_var_139112_sub_indel_2     GT      G       .       P
    MTR5065 .       A       AC      5808.74 MEAN_ALLELE_BALANCE;MQ  AC=2;ACMI
    MTR5065 UNION_BC_k31_var_139112_sub_indel_3     A       AC      .       P
    MTR5084 UNION_BC_k31_var_139112_sub_indel_4     AT      A       .       P
    MTR5155 .       TAC     T       449.65  DP;MQ;QD        AC=3;ACMI=3;ACMJ=

Actually loading the data doesn't look too bad:
 http://gmod.org/wiki/JBrowse_FAQ#How_do_I_set_up_a_VCF_file.3F

**Comment by kimrutherford** [comment]
> (although a link from JBrowse to the appropriate SNP in the DB will do for this meta data I have no idea how this connection is made)

I'm not sure how to do that either.  It will be easier to work out once we have a VCF track loaded.

**Comment by kimrutherford** [comment]
> We'll need to fix the chromosome IDs in the VCF files before we can load them into JBrowse. It's currently I, II, III etc.

Note to self, this issue has the new and old IDs:
  https://github.com/pombase/website/issues/671

**Comment by kimrutherford** [comment]
> I have a feeling this will be the complicated part.... I * think*, but I am not sure, that there must be
> IDs assigned for individual SNPs.

It looks like Ensembl assign an ID for each variant.  Those IDs aren't in the VCF file so won't be available in JBrowse.

> Presumably Ensembl have an easy way to access these based on sequence

Could you explain that?  I don't understand.

**Comment by ValWood** [comment]
I don't think Ensembl assigns them, I think the variation database assigns them.....and Ensembl use them...

**Comment by ValWood** [comment]
I'll have a poke around

**Comment by ValWood** [comment]
Maybe we just link to this?
https://www.ebi.ac.uk/eva/?Variant%20Browser
Although I can't find the pombe data?

In true EBI style they have found some interesting organisms  the lesser known "500 V1" and the  "5.0.1"

**Comment by ValWood** [comment]
ffs from here they do have "fission yeast" as an option
https://www.ebi.ac.uk/eva/?Study-Browser&browserType=sgv
but  checking the boxes does not appear to filter anything????

**Comment by kimrutherford** [comment]
Midori found lots of files on the EBI FTP site.  Which ones do we need to display?  Here's the list:

```
all.pi.wig
chr1.pi.wig
chr2.pi.wig
chr3.pi.wig
LTR-insertions.bed
mito.pi.wig
Spombe.2013-01-02.filt3c.nr57-final.snps.anno-snpeff3.vcf.gz
Spombe.indels.unique.nr57.final.annotated.2014-01-23.vcf.gz
```

We also have two other files that they sent in the big dump of all the Ensembl code and data:

```
DJeffares_Diversity.bw
LTR-insertions.PMID.25665008.bed
```

**Comment by kimrutherford** [comment]
> Midori found lots of files on the EBI FTP site. Which ones do we need to display? Here's the list:

I'm going to need help with this.

**Comment by Antonialock** [comment]
is this helpful? https://www.ebi.ac.uk/ena/data/view/PRJEB2733

**Comment by ValWood** [comment]
wow, there is no indication what that data is whatsoever.

F no
A no
I no
R no

Great proliferation  of identifiers though..... (five per sample)

**Comment by Antonialock** [comment]
I clicked on the "analysis files" tab.

there are 6 files listed (VCF)

ERZ289252 S. pombe indels MTR
ERZ289253 S. pombe SNPs MTR
ERZ289254 S. pombe indels
ERZ289255 S. pombe SNPs
ERZ289256 S. pombe SNPs AB325691
ERZ289257 S. pombe indels AB325691

it looks like they bundled all indels into one track, and all snps into one track, and then had one track for everything

![screen shot 2018-06-13 at 23 38 48](https://user-images.githubusercontent.com/12727506/41382195-0401cec4-6f63-11e8-85e8-3b501ddbd44c.png)

**Comment by Antonialock** [comment]
adding track descriptions for these.

@djeffares in the paper you report the accession numbers in dbSNP for the SNPs 974514578-974688138
but this ID is for a human gene?  https://www.ncbi.nlm.nih.gov/snp/?term=974688138

Also are you happy with these track labels for the 3 tracks or can they be tweaked?


Indels and SNPs in 57 non-clonal S. Pombe strains
Indels in 57 non-clonal S. Pombe strains
SNPs in 57 non-clonal S. Pombe strains

**Comment by Antonialock** [comment]
Ah, I also found this @kimrutherford from an email exchange April 17 between all of us

from Dan:
"Thanks for looking into this.  I put a new set of SNP, indel and structural variation calls in vcf format on figshare here: https://figshare.com/projects/fission_yeast_structural_variation/15798"

**Comment by djeffares** [comment]
Hi all,

Yes, this is the best source. These are variant calls for the 161 strains though (not just the 57 non clonal). So its a better, more complete data set.

For @kimrutherford <https://github.com/kimrutherford> 

Indels are here:
https://figshare.com/s/bf9075c15a9b481e2001 <https://figshare.com/s/bf9075c15a9b481e2001>

SNPs are here:
https://figshare.com/s/5e556e4698b128181cb8 <https://figshare.com/s/5e556e4698b128181cb8>

Thanks! Let me know if problems come up.

cheers,
Dan



On 15 Jun 2018, at 08:14, Antonia <notifications@github.com> wrote:

Ah, I also found this @kimrutherford <https://github.com/kimrutherford> from an email exchange April 17 between all of us

from Dan:
"Thanks for looking into this. I put a new set of SNP, indel and structural variation calls in vcf format on figshare here: https://figshare.com/projects/fission_yeast_structural_variation/15798 <https://figshare.com/projects/fission_yeast_structural_variation/15798>"

**Comment by Antonialock** [comment]
Ok, removed the line for the combined track and updated the descriptions to 

Indels in 161 S. pombe natural isolates
SNPs in 161 S. pombe natural isolates

**Comment by kimrutherford** [comment]
The two new tracks are live:
  https://www.pombase.org/jbrowse/?loc=II%3A2007265..2008224&tracks=PomBase%20forward%20strand%20features%2CDNA%20sequence%2CPomBase%20reverse%20strand%20features%2CIndels%20in%20161%20S.%20pombe%20natural%20isolates%2CSNPs%20in%20161%20S.%20pombe%20natural%20isolates&highlight=

Sorry for the delay Dan.  Please let us know if you spot any problems.

**Comment by djeffares** [comment]
Looks great - thanks Kim!

cheers,
Dan








On 21 Jun 2018, at 12:42, Kim Rutherford <notifications@github.com> wrote:

The two new tracks are live:
https://www.pombase.org/jbrowse/?loc=II%3A2007265..2008224&tracks=PomBase%20forward%20strand%20features%2CDNA%20sequence%2CPomBase%20reverse%20strand%20features%2CIndels%20in%20161%20S.%20pombe%20natural%20isolates%2CSNPs%20in%20161%20S.%20pombe%20natural%20isolates&highlight= <https://www.pombase.org/jbrowse/?loc=II%3A2007265..2008224&tracks=PomBase%20forward%20strand%20features%2CDNA%20sequence%2CPomBase%20reverse%20strand%20features%2CIndels%20in%20161%20S.%20pombe%20natural%20isolates%2CSNPs%20in%20161%20S.%20pombe%20natural%20isolates&highlight=>
Sorry for the delay Dan. Please let us know if you spot any problems.

**Comment by kimrutherford** [comment]
> In the ensembl browser.....different types of variant have different colour...

Sorry I didn't see that.  We can colour features in JBrowse but I don't know of a way to add a legend so it might not be useful.

**Comment by Antonialock** [comment]
Kim do you think this is possible?

**Comment by kimrutherford** [comment]
I'm a bit confused.  We have "Type   insertion" in the details for the variation features.  Do we need more?

Perhaps we can chat about this at tomorrow's meeting?

**Comment by ValWood** [comment]
Antonia can you open a new ticket for what is outstanding, and close this one? I'm a bit confused too.

**Comment by Antonialock** [comment]
Val said:

> In the ensembl browser.....different types of variant have different colour...

Kim replied:

> Sorry I didn't see that. We can colour features in JBrowse but I don't know of a way to add a legend so it might not be useful.

then I linked to this ticket with some screenshots:
GMOD/jbrowse#937

**Comment by ValWood** [comment]
If we never announced the variation data availability in JBrowse, we should do that, maybe after this tweak

**Comment by ValWood** [comment]
@Antonialock  what is to do here to announce this data?

**Comment by Antonialock** [comment]
Colour variants by type if possible

it seems like it will be a pain. Dataset was already announced, this was just a tweak.

**Comment by ValWood** [comment]
ok if there is another variation dataset, can maybe implement then...
